mm2src/coins/z_coin.rs

```diff
@@ -394,18 +401,128 @@ impl ZCoin {
         let total_output = big_decimal_from_sat_unsigned(total_output_sat, self.utxo_arc.decimals);
         let total_required = &total_output + &tx_fee;
 
-        let spendable_notes = self
-            .spendable_notes_ordered()
-            .await
-            .map_err(|err| GenTxError::SpendableNotesError(err.to_string()))?;
+        let mut retries = 0;
+
+        loop {
+            // Wait for block sync and previous tx confirmation.
+            let sync_guard = self.wait_for_gen_tx_blockchain_sync().await?;
+            let spendable_notes = self
+                .spendable_notes_ordered()
+                .await
+                .map_err(|err| GenTxError::SpendableNotesError(err.to_string()))?;
+
+            let change_notes = self.z_fields.locked_notes_db.load_all_notes().await?;
+            println!("LOCKED NOTES: {change_notes:?}");
+            if change_notes.is_empty() {
+                return self
+                    .gen_tx_impl(
+                        t_outputs,
+                        z_outputs,
+                        spendable_notes.iter(),
+                        total_required,
+                        tx_fee,
+                        sync_guard,
+                    )
+                    .await;
+            }
+
+            {
+                // check if remaining notes will be enough for this tx and skip waiting,
+                let change_notes_rseeds = change_notes.iter().map(|n| n.rseed.clone()).collect::<HashSet<_>>();
```

Comment:
Looks like there is code duplication here and in my_balance fn, where we filter unlocked notes.
I think we could create a new function get_spendable_unlocked_notes which would do such filtering and use it everywhere.

Reply:
I think we can do without this it's just 2-3 lines of codes and they both return different values